ragna/source_storages/_embedding_model.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from sentence_transformers import SentenceTransformer
+import torch
+
+from ragna.core._components import Component
+
+from ragna.source_storages._vector_database import Chunk
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+class Embedding:
+    embedding: list[float]
+    chunk: Chunk
+
+    def __init__(self, embedding: list[float], chunk: Chunk):
+        super().__init__()
+        self.embedding = embedding
+        self.chunk = chunk
+
+
+class GenericEmbeddingModel(Component, ABC):
+    _EMBEDDING_DIMENSIONS: int
```

Comment:
Shouldn't this be an attribute on the `Embedding` class? Or maybe we can eliminate it all together since it is just `len(Embedding().embedding)`?

Reply:
No, because it specified the embedding dimensions of the model.  Models create an embedding of N dimensions.  But yes I will just remove it entirely.